src/yaml/schema/core/time_parser.cr

```diff
@@ -109,7 +109,7 @@ struct YAML::Schema::Core::TimeParser
 
     return nil if @reader.has_next?
 
-    time = new_time(year, month, day, hour, minute, second, millisecond: millisecond)
+    time = new_time(year, month, day, hour, minute, second, nanosecond: millisecond * 1_000_000)
```

Comment:
I know this is not a good way to set milliseconds, should we allow a `Time::Span` to be passed, so we would write: `(..., nanosecond: millisecond.milliseconds)` ?

Reply:
Why should this not be a good way? Multiplying by `1_000_000` is the proper transformation from milli do nano.

Reply:
yes maybe, maybe I just don't like to see some numeric constants written in code like that, but it's probably totally fine actually.